exercises/concept/amusement-park/amusement-park.js

```diff
@@ -0,0 +1,67 @@
+// @ts-check
+
+/**
+ * Creates a new visitor.
+ *
+ * @param {string} name
+ * @param {number} age
+ * @param {string} ticketId
+ * @returns {Object} visitor
+ * @returns {string} visitor.name
+ * @returns {number} visitor.age
+ * @returns {string} visitor.ticketId
+ */
+export function createVisitor(name, age, ticketId) {
+  throw new Error('Please implement the createVisitor function.');
+}
+
+/**
+ * Revokes a ticket for a visitor.
+ *
+ * @param {Object} visitor
```

Comment:
This would then become:

```javascript
/**
 * @param {Visitor} visitor the visitor with an active ticket
 * @returns {Visitor} the visitor without a ticket
 */
```

We could _type_ that better, but that would give away the answer, maybe.

e.g.

```typescript
type Visitor = ActiveVisitor | PastVisitor

type ActiveVisitor = { name: string, age: string, ticketId: string }
type PastVisitor = { name: string, age: string, ticketId: null }
``` 

Then it would be:

```javascript
/**
 * @param {ActiveVisitor} visitor the visitor with an active ticket
 * @returns {PastVisitor} the visitor with their ticket revoked
 */
```

Reply:
I think the Active vs. Past version is a bit too much. I went with just Visitor.